repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/AuthPolicyPreProcessor.java

```diff
@@ -165,6 +165,36 @@ private void processCreatePolicy(AtlasStruct entity) throws AtlasBaseException {
         RequestContext.get().endMetricRecord(metricRecorder);
     }
 
+    private void validateAndReduce(AtlasEntity policy) {
+        ArrayList<String> resources = (ArrayList<String>) policy.getAttribute(ATTR_POLICY_RESOURCES);
+        boolean hasAllDomainPattern = false;
+
+        for (int i = 0; i < resources.size(); i++) {
+            String[] resourceParts = resources.get(i).split(":");
+            String resource = resourceParts[resourceParts.length - 1];
+            if (resource.equals("*") || resource.equals("*/super") || resource.equals("default/domain/*/super")) {
+                resources.remove(i);
+                hasAllDomainPattern = true;
+                i--;
+            }
+        }
```

Comment:
rewrite this with simple: 
hasAllDomainPattern = resources.contains("entity:\*") || 
resource.contains("entity:\*/super") || 
resource.contains("entity:default/domain/*/super")